base/pkg/entry.jl

```diff
@@ -50,19 +50,20 @@ function add(pkg::AbstractString, vers::VersionSet)
     outdated = :maybe
     @sync begin
         @async if !edit(Reqs.add,pkg,vers)
-            ispath(pkg) || throw(PkgError("unknown package $pkg"))
```

Comment:
Stack trace is useful if you want to dig into why Pkg works the way it does when the error gets thrown, even if it's something seemingly obvious like a package that doesn't exist. It also shouldn't silently succeed here.